sys/include/xtimer/tick_conversion.h

```diff
@@ -29,16 +29,142 @@ extern "C" {
 #endif
 
 /* Some optimizations for common timer frequencies */
+
+/*
+ * First, check if XTIMER_HZ is multiple of 921600Hz
+ * This is a common frequency for baud rate generation. We use the fact that
+ * the greatest common divisor between 921600 and 1000000 is 1600, so instead of
+ * multiplying by the fraction (921600 / 1000000), we will instead use
+ * (576 / 625), which reduces the truncation caused by the integer widths
+ */
+#if (XTIMER_HZ % 576 == 0)
+/*
+ * Second, check if its using shifting
+ */
 #if (XTIMER_SHIFT != 0)
+/*
+ * If it uses shifting, just check if it's not multiple of 576, if so,
+ * throw an error
+ */
+#if (XTIMER_HZ % 576 != 0)
```

Comment:
This will never be true, as you check above the opposite, or do I miss something?